### PR TITLE
use transform for affine transformations in raster packed

### PR DESCRIPTION
**Pull request by sebastianbaginski**
Use `transform` passed from `DrawState` for affine transformations on raster images. Added tests.
@marcusnaslund review

### CODE REVIEW

File: source/draw/RasterPacked.ooc [hunk 1]
```diff
@@ -38,13 +38,15 @@ RasterPacked: abstract class extends RasterImage {
 		this _buffer = null
 		super()
 	}
-	_resizePacked: func <T> (sourceBuffer: T*, source: This, sourceBox, resultBox: IntBox2D, transform: FloatTransform3D, interpolate, flipX, flipY: Bool) {
-		if (this size == source size && this stride == source stride && sourceBox == resultBox && sourceBox size == source size && sourceBox leftTop x == 0 && sourceBox leftTop y == 0 && !flipX && !flipY)
+	_resizePacked: func <T> (sourceBuffer: T*, source: This, sourceBox, resultBox: IntBox2D, transform: FloatTransform3D, interpolate, flipX, flipY: Bool, nullPixel: T) {
+		version(safe)
+			raise(transform determinant equals(0.0f), "invalid transform in _resizePacked !")
```
Comment (marcusnaslund) [inline source/draw/RasterPacked.ooc]: Extra space before `!`

File: test/draw/RasterMonochromeTest.ooc [hunk 2]
```diff
@@ -124,6 +124,16 @@ RasterMonochromeTest: class extends Fixture {
...
+			transform := FloatTransform3D createTranslation(source width / 2, source height / 2, 0.0f) * FloatTransform3D createRotationZ(3.14 / 7) * FloatTransform3D createTranslation(-source width / 2, -source height / 2, 0.0f)
+			DrawState new(target) setInputImage(source) setTransformNormalized(transform) setInterpolate(true) draw()
+			output := "test/draw/output/RasterMonochromeTest_rotated.png"
+			target save(output)
+			(target, source) referenceCount decrease()
+			output free()
+		})
 		/*this add("distance, convertFrom RasterRgba", func {
 			source := this sourceFlower
 			output := "test/draw/output/RasterRgbToMonochrome.png"
```
Comment (marcusnaslund) [inline test/draw/RasterMonochromeTest.ooc]: What's the story behind this test?

File: test/draw/RasterMonochromeTest.ooc [hunk 3]
```diff
@@ -124,6 +124,16 @@ RasterMonochromeTest: class extends Fixture {
 			expect(image[0, 2] y as Int, is equal to(value as Int))
 			image referenceCount decrease()
 		})
+		this add("rotate (Z)", func {
+			source := RasterMonochrome open(this sourceFlower)
+			target := RasterMonochrome new(source size)
+			transform := FloatTransform3D createTranslation(source width / 2, source height / 2, 0.0f) * FloatTransform3D createRotationZ(3.14 / 7) * FloatTransform3D createTranslation(-source width / 2, -source height / 2, 0.0f)
```
Comment (marcusnaslund) [inline test/draw/RasterMonochromeTest.ooc]: `createRotationZ(3.14f / 7)`

File: test/draw/RasterMonochromeTest.ooc (hunk 2 above)
Comment (sebastianbaginski) [inline test/draw/RasterMonochromeTest.ooc]: Good question, but I have no idea. I re-enabled it with increased tolerance.

**Comment by marcusnaslund** [comment]
Oh, memplay failed because of my recent changes (now, but not before...?)
Will fix

**Comment by marcusnaslund** [comment]
Memory leak fixed, restarting tests.

**Comment by marcusnaslund** [comment]
All green now.

**Pull request merged**